src/components/program/SessionDetails.tsx

```diff
@@ -0,0 +1,85 @@
+'use client';
+
+import { dayAndTimeFormatWithMonth } from "../../utils/dateformat.ts";
+import type { Session } from "../../types/program.ts";
+import {useEffect, useState} from "react";
+
+interface Props {
+    session: Session;
+}
+
+export const SessionDetails = ({ session }: Props) => {
+    const [isFavorite, setIsFavorite] = useState(false);
+
+    useEffect(() => {
+        const favoriteIds = JSON.parse(localStorage.getItem("favorites") || "[]");
+        setIsFavorite(favoriteIds.includes(session.id));
+    }, [session.id]);
```

Comment:
Dette trenger du ikke en useEffect for, dette kan du heller legge inn i useState sin [initializer-funksjon](https://react.dev/reference/react/useState#avoiding-recreating-the-initial-state)

Reply:
f652f008